### PR TITLE
Adds nimbus feature manifest to iOS build steps

### PR DESCRIPTION
**Pull request by tarikeshaq**
~~**IMPORTANT**: This should not merge until firefox-ios is using appservices v91.0.0 or higher~~ We're already on 91.0.1 😄 

Fixes https://mozilla-hub.atlassian.net/browse/EXP-1983

Adds downloading, and using the nimbus fml as a build step. For background info on the feature manifest language, https://experimenter.info/fml-spec

TODOs:
- [x] Verify that how I handle channels/schemes is correct
- [ ] Verify that the FML generation works okay for people running M1s 
- [x] Verify my assumption that this will only run on macs is OK
- [x] Clean up the `nimbus.fml.yaml`. right now it's mostly a draft and the default values are most likely wrong
     - Further clean-up can happen once the project is migrating to use the FML, which should be possible as soon as this lands
- [ ] Have a couple of people run this on their machines to make sure local development is OK

cc @jhugman

### CODE REVIEW

File: bin/nimbus-fml.sh
Comment (jhugman) [inline bin/nimbus-fml.sh:145]: I think we should standardize the names we use between Android and iOS; this will make it a bit easier to document.

Suggest: could we go with names:

* `FxNimbus` as the class name.
* `FxNimbus.swift` as the filename.
* `nimbus.fml.yaml` for the manifest name.

File: bin/nimbus-fml.sh
Comment (jhugman) [inline bin/nimbus-fml.sh:88]: I got confused about the FML path. You mean it as `MANIFEST_PATH` rather than `NIMBUS_FML_BINARY_PATH`.

File: bin/nimbus-fml.sh
Comment (jhugman) [inline bin/nimbus-fml.sh:31]: :)

File: bin/nimbus-fml.sh
Comment (jhugman) [inline bin/nimbus-fml.sh:139]: Is there room here for local development?

**Review by jhugman — COMMENTED** [review]
Quick drive by, since it's still  in draft. 

TLDR some naming things, but otherwise, it's looking nice.

File: bin/nimbus-fml.sh
Comment (tarikeshaq) [reply bin/nimbus-fml.sh]: Hmm yeah - this will be annoying to keep track of (and we will 100% forget to upgrade it if we don't introduce breaking changes.. we can have the script try to extract the version out of the xcode project, but that will be fragile 😕

File: bin/nimbus-fml.sh
Comment (tarikeshaq) [reply bin/nimbus-fml.sh:145]: On non-mac devices? Not for now - I'll file it as a follow-up since it'll be hard to verify if the FML generated swift that builds OK on a non-mac device that can't build the project 😞

**Comment by tarikeshaq** [comment]
cc @nishant2718 If you happen to have an M1, do you mind pulling this PR and giving it a go on a simulator? otherwise, I think this is almost landable and we can start migrating stuff to use it

**Comment by tarikeshaq** [comment]
Also, it would be great if we can get a FirefoxBeta build going, I want to make sure this builds OK for non-fennec builds before landing

**Comment by tarikeshaq** [comment]
Looks the bitrise build is broken, and its related... looking into it

**Comment by nishant2718** [comment]
@tarikeshaq I don't have an M1, but I can get someone who does to test it locally. Would you like for us to do it now, or after BR passes?

File: bin/nimbus-fml.sh
Comment (jhugman) [reply bin/nimbus-fml.sh:145]: ah no, sorry. I mean local development of Nimbus and the FML.

**Comment by tarikeshaq** [comment]
Thanks a lot @nishant2718! Ya I think we can get someone with an M1 to give building it a shot now, and is it possible to re-trigger bitrise? I **think** I fixed the issue 😄

File: nimbus.fml.yaml
Comment (jhugman) [inline nimbus.fml.yaml:59]: Question for @nishant2718 : should these all be true now?

**Comment by tarikeshaq** [comment]
I think that bitrise is failing for [unrelated](https://github.com/mozilla-mobile/firefox-ios/pull/9905/checks?check_run_id=5116467349) reasons 😄

File: Client.xcodeproj/project.pbxproj
Comment (tarikeshaq) [inline Client.xcodeproj/project.pbxproj:689]: huh this is not right, probably a side-effect to the merge conflict, I'll fix it

Comment (lmarceau) [reply Client.xcodeproj/project.pbxproj:689]: Mh 🤔 @electricRGB had the same change in his PR here: https://github.com/mozilla-mobile/firefox-ios/pull/9970
I guess somehow it should never had been there in the first place?

Comment (tarikeshaq) [reply Client.xcodeproj/project.pbxproj:689]: Actually, `RUSTPlaces.frameworks` doesn't exist anymore, kinda surprised it's still in the `.pbxproj`

@nbhasin2 (since you have the context about the frameworks and all 😬), **xcode automatically did this for me**, is there an xcode setting to force it to remove the framework from here, or is it safe to let it do this "null" thing it's doing here

Comment (tarikeshaq) [reply Client.xcodeproj/project.pbxproj:689]: @lmarceau we commented at the same time 😆 Ya I don't think it should be there

Comment (lmarceau) [reply Client.xcodeproj/project.pbxproj:689]: @tarikeshaq haha yeah, and same conclusion 😝

Comment (electricRGB) [reply Client.xcodeproj/project.pbxproj:689]: It is now gone in `main` :)

File: nimbus.fml.yaml
Comment (nishant2718) [reply nimbus.fml.yaml:59]: I'm not sure what `recent-explorations` is to be honest 😅 . We can probably keep that false for all channels for now. 

In Prod, there's currently an experiment running that enables Jump Back In and Recently Saved. In Beta, both of those are default off.

So maybe we need two additional channel defaults for Prod and Beta that reflect that? Where Prod and Beta would have those two off (and the rest on).

Comment (tarikeshaq) [reply nimbus.fml.yaml:59]: FWIW, the values here are what I set as I was going over what experiments are in use in iOS - we'll need to make sure they're correct when iOS switches its experiments to the FML completely - but for now, having the FML should unblock the messaging work

I'll follow up with another PR to switch the experiments we have to use the FML, and we can correct those values there

**Review by nishant2718 — APPROVED** [review]
LGTM - Going to merge this onto a feature branch and test if we can produce a nightly first. THEN, merge into main if all goes well.

**Pull request merged**
